### PR TITLE
Fix password reset when using email as login

### PR DESCRIPTION
**Pull request by Hrittik20**
Fixes: #943

### CODE REVIEW

File: src/plone/restapi/services/users/add.py
Comment (stevepiercy) [inline src/plone/restapi/services/users/add.py:344]: This needs to pass lint, specifically black is failing: https://github.com/plone/plone.restapi/actions/runs/8472294504/job/23214126598?pr=1767

Also I don't know whether this continues to allow use of the username, as well as the email. In fact, I don't think this is the correct fix for this, as the previous line requires a user to be authenticated.

**Review by stevepiercy — DISMISSED** [review]

**Review by stevepiercy — COMMENTED** [review]
This still needs review by a core contributor.

I wonder whether we should have a unit test for both email and username? But let's see what happens in review.

File: src/plone/restapi/services/users/add.py
Comment (stevepiercy) [reply src/plone/restapi/services/users/add.py]: I am not entirely sure. I think we need another endpoint, or adapt an existing one, or work with the permissions, to do what we need in Volto, because this endpoint requires authentication. I haven't thought through the details.

**Review by davisagli — APPROVED** [review]
Looks right to me.

You haven't followed the instructions from the bot to start the full Plone test suite with this change. I'll do that now.

**Comment by davisagli** [comment]
> You haven't followed the instructions from the bot to start the full Plone test suite with this change.

Not your fault, the bot didn't post the message it usually does for some reason. Maybe because this PR was submitted from a fork instead of from a branch inside the plone/plone.restapi repository.

**Comment by davisagli** [comment]
@jenkins-plone-org please run jobs

**Pull request merged**
